Automattic/wp-calypso
# Migrate the requestPluginKeys data getter to a plain wpcom.req.get

The `requestPluginKeys` getter was used only in the Jetpack Product Installer screen. This PR migrates it to a plain `wpcom.req.get` request that sets local state (`this.state.pluginKeys`). It's an one-off request, it doesn't need any caching or sharing between components, so not even a basic React Query would add value.

**How to test:**
@tyxla you implemented this screen in #31684 so you're the best person to test this change I guess 🙂 

However, today with the new Jetpack plans it's very difficult to get to this screen. I'm not sure if it's needed at all and whether we can remove it?

How to get there? First you need to buy a legacy Jetpack plan. Do that by going to:
```
/checkout/jetpack/:site/jetpack_personal?flow=coming_from_login&purchasetoken=foo
```
It needs to be a Jetpack Checkout because a normal `/checkout` would detect a legacy plan and would redirect to a screen that recommends one of the new plans (see testing instructions for #60480)

The `flow` and `purchasetoken` params are needed to make the Checkout code think it's really `isJetpackCheckout`. Without them, the product wouldn't be added to the cart.

Now this adds Jetpack Personal to your cart but **don't pay for it yet**! Because we're in Jetpack Checkout, the thank-you page would be different.

You need to abandon this checkout and go to `/checkout/:site`. The legacy plan is still in your cart, but now you're checking out with `isJetpackCheckout = false`. Pay for the plan. It will probably fail because the `jetpackPurchaseToken` is invalid. I had to disable this check on server in `StoreProduct::pre_purchase` and sandbox my `public-api`.

Now, finally, after finishing the payment, you're taken to `/plans/my-plan/:site?install=all` and the product install check is performed.

You can also test this page by going to it directly. If you previously deactivate and delete the Akismet plugin, the page should install it again.
